Makefile.kokkos

```diff
@@ -812,11 +827,29 @@ endif
 ifeq ($(KOKKOS_INTERNAL_USE_ARCH_ARMV9_GRACE), 1)
   tmp := $(call kokkos_append_header,"$H""define KOKKOS_ARCH_ARMV9_GRACE")
   tmp := $(call kokkos_append_header,"$H""define KOKKOS_ARCH_ARM_NEON")
+  tmp := $(call kokkos_append_header,"$H""define KOKKOS_ARCH_ARM_SVE")
 
   KOKKOS_CXXFLAGS += -mcpu=neoverse-v2  -msve-vector-bits=128
   KOKKOS_LDFLAGS += -mcpu=neoverse-v2  -msve-vector-bits=128
 endif
 
+ifeq ($(KOKKOS_INTERNAL_USE_ARCH_ARMV84_SVE), 1)
+  tmp := $(call kokkos_append_header,"$H""define KOKKOS_ARCH_ARMV84_SVE")
+  tmp := $(call kokkos_append_header,"$H""define KOKKOS_ARCH_ARM_NEON")
+  tmp := $(call kokkos_append_header,"$H""define KOKKOS_ARCH_ARM_SVE")
+
+  # try_run() to query HW SVE vector-length
+  tmp := $(strip $(shell mktemp))
+  KOKKOS_INTERNAL_ARCH_ARM_SVE_VL := $(strip $(shell $(CXX) -march=armv8.4-a+sve \
+                                                     $(KOKKOS_PATH)/cmake/compile_tests/get_sve_hw_vl.cpp \
+                                                     -o $(tmp) > /dev/null 2>&1 \
+                                                     && $(tmp) | sed 's/SVE_HW_VL=//' \
+                                                     ; rm -f $(tmp) ))
+
+  KOKKOS_CXXFLAGS += -march=armv8.4-a+sve -msve-vector-bits=$(KOKKOS_INTERNAL_ARCH_ARM_SVE_VL)
+  KOKKOS_LDFLAGS += -march=armv8.4-a+sve -msve-vector-bits=$(KOKKOS_INTERNAL_ARCH_ARM_SVE_VL)
```

Comment:
I am not aware of any Armv8.4-A CPU with SVE support that uses a vector width different than 256 (because these are basically Neoverse V1-based, anyway). Maybe this could be simplified as:
```suggestion
  KOKKOS_CXXFLAGS += -march=armv8.4-a+sve -msve-vector-bits=256
  KOKKOS_LDFLAGS += -march=armv8.4-a+sve -msve-vector-bits=256
```

Alternatively, it might be simpler to redefine `Kokkos_ARCH_ARMV84_SVE` as `Kokkos_ARCH_NEOVERSE_V1`to use an Arm _microarchitecture_ instead of a generic architecture. It is kind of what Kokkos already does with, e.g., A64FX (which is its own microarchitecture implementing Armv8.2-A+SVE).

Reply:
I agree that Neoverse-V1 is today the only one in Armv8.4-a+SVE(1) config, and SVE1 technically does not prohibit any 512, 1024 or 2048-bit.

In this PR, we aimed to support any Armv8.4-a+SVE (not necessarily Neoverse-V1). This VL-detection mechanism is used for both Grace (128-bit) and Graviton3 (256-bit), and can be used for A64FX, too, as it does not cost much to check for the real HW SVE VL at configure.